concent_api/gatekeeper/views.py

```diff
@@ -124,73 +161,193 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
+            client_public_key,
+        )
     if any((len(file["checksum"]) == 0 or file["checksum"].isspace()) for file in loaded_golem_message.files):
-        return gatekeeper_access_denied_response("'checksum' cannot be blank or contain only whitespace.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'checksum' cannot be blank or contain only whitespace.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_EMPTY.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     if not all(":" in file["checksum"] for file in loaded_golem_message.files):
-        return gatekeeper_access_denied_response("'checksum' must consist of two parts separated with a semicolon.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'checksum' must consist of two parts separated with a semicolon.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_FORMAT.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     file_checksums = [tuple(file["checksum"].split(":")) for file in loaded_golem_message.files]
     if not set(checksum_type for checksum_type, checksum in file_checksums).issubset(set(HashingAlgorithm._value2member_map_)):  # type: ignore
-        return gatekeeper_access_denied_response("Unrecognized checksum type.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "Unrecognized checksum type.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_UNRECOGNIZED_TYPE.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     assert set(HashingAlgorithm) == {HashingAlgorithm.SHA1}, "If you add a new hashing algorithms, you need to add validations below."
     if any(VALID_SHA1_HASH_REGEX.fullmatch(file_checksum[1]) is None for file_checksum in file_checksums):
-        return gatekeeper_access_denied_response("Invalid SHA1 hash.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "Invalid SHA1 hash.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_SHA1_HASH.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     file_sizes = [file["size"] for file in loaded_golem_message.files]
     if any((file_size is None) for file_size in file_sizes):
-        return gatekeeper_access_denied_response("'size' must be an integer.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'size' must be an integer.",
+            ErrorCode.MESSAGE_FILES_SIZE_WRONG_TYPE.value,
```

Comment:
Maybe `MESSAGE_FILES_SIZE_EMPTY`? We discern between "wrong type" and "empty" for the checksum field. They should be handled the same way. So either both always use "wrong type" or both discern between these two situations.

Reply:
:+1: